lib/scanner/walk_test.go

```diff
@@ -366,6 +367,36 @@ func TestWalkSymlinkWindows(t *testing.T) {
 	}
 }
 
+func TestWalkRootSymlink(t *testing.T) {
+	// Create a folder with a symlink in it
+	tmp, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmp)
+
+	link := tmp + "/link"
+	dest, _ := filepath.Abs("testdata/dir1")
+	if err := os.Symlink(dest, link); err != nil {
```

Comment:
Why don't you use `DebugSymlinkForTestsOnly` anymore?

Reply:
Apart from this, the PR looks good to me.

Reply:
That's a mistake, I had misremembered @calmh's open issue for Go https://github.com/golang/go/issues/22874 as a Syncthing issue that Go now supported Windows symlinks.

Reply:
Using `os.Symlink` was definitively a bug and shows that the issue could still be in scanner. `os.Symlink` was triggering the `t.Skip` case, switching back to `DebugSymlinkForTestsOnly` resurfaces that scanner is not detecting the files.

@AudriusButkevicius is there a reason an err on creating symlinks on Windows shouldn't trigger a failure? I know it's that way in config_test.go's `TestFolderCheckPath` and prior walk_test.go tests. Is this an issue for testing on older versions of Windows?